src/compas/files/base_reader.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+from __future__ import division
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+class BaseReader(object):
+    """Base class containing file reading functions for file extension specific readers
+
+    Attributes
+    ----------
+    location : str or pathlib object
+        Path or URL to the file
+    """
+
```

Comment:
should the base reader not define the interface for default methods of the other readers as well?
* `read`
* ...

Reply:
with a `NotImplementedError`

Reply:
Sure!

Reply:
I'd say `read` is the only shared function, maybe `is_valid` could be implemented for all. `is_binary` seems a bit silly for ascii only file formats.

Added to end of file.